Corrected offests transformation in model.Differ when multiple change items interfere with each other

scofalik opened the pull request:
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Fix: Corrected offests transformation in `model.Differ` when multiple change items interfere with each other. Closes #1309. Closes https://github.com/ckeditor/ckeditor5/issues/849.

pjasiun commented on src/model/differ.js:
```diff
@@ -547,6 +547,8 @@ export default class Differ {
 	 * @param {Array.<Object>} changes An array containing all the changes done on that element.
 	 */
 	_handleChange( inc, changes ) {
+		inc.newHowMany = inc.howMany;
```
Some comment what `newHowMany` is and why it is needed is needed.

pjasiun commented:
Missing tests.

Mgsy commented:
After switching to this branch, there is the error in the console on init:
```
article.js:24 TypeError: editor.conversion.elementToElement is not a function
    at BlockQuoteEngine.init (blockquoteengine.js:42)
    at <anonymous>
```

scofalik commented:
@pjasiun I will add a test.
@Mgsy are you sure you have all branches on `master`?

Mgsy commented:
>@Mgsy are you sure you have all branches on master?

Yes, all branches are on `master`. I've removed all packages and installed it once again and the problem still exists.

scofalik commented:
First, I had to refactor all the tests, because they were incorrect after engine refactoring. They worked by a coincidence.

Other than that, I rebased this branch with `master` and set all other branches on current `master`s. Tests are all green. I also added the missing test for this case.

Mgsy commented:
The crash is gone, but data is restored incorrectly:

![bug_ckeditor5](https://user-images.githubusercontent.com/20988892/36425685-d14fd9d6-1647-11e8-80bf-59a55fb0fc13.gif)

It will be a part of this PR or should I report it in the [Undo umbrella](https://github.com/ckeditor/ckeditor5-undo/issues/65)?

scofalik commented:
Yes, I already reported the bug in the undo ticket.
